hw/dv/tools/dvsim/questa.hjson

```diff
@@ -0,0 +1,114 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  build_cmd:  "vlib work && {job_prefix} vlog"
+  run_cmd:    "{job_prefix} vsim"
+
+  // Indicate the tool specific helper sources - these are copied over to the
+  // {tool_srcs_dir} before running the simulation.
+
+  build_opts: ["-sv",
+               "-timescale 1ns/1ps",
+               "+incdir+/${QUESTA_HOME}/verilog_src/uvm-1.2/src/",
+               "/${QUESTA_HOME}/verilog_src/uvm-1.2/src/uvm_pkg.sv",
+               // Multi-file compilation unit option for Questa
+               // Needed to use macros across CU-s
+               "-mfcu"
+               "-f {sv_flist}"]
+
+  run_opts:   ["-sv_seed {seed}",
+               // List multiple tops for the simulation
+               "{sim_tops}",
+               // dv_macros.svh has a macro printing null using %0d
+               // format specifier, this is illegal as per strict LRM
```

Comment:
I am afraid I don't have time for that search, will modify the comments instead :-)

Reply:
I don't think that the spec (IEEE 1800-2017) requires `%d` to support `null` operands, nor does it disallow it. The relevant paragraph is probably in section 21.2.1.2, just after the table. Specifically, it says that "*The integer format specifiers, %h , %x , %d , %o , %b , %c , %u , and %z (uppercase or lowercase), may be used with any of the SystemVerilog integral data types, including enumerated types and packed aggregate data types.*".

Since `null` isn't a member of any integral type, this doesn't require the simulator to support printing it with `%d` (nor does it forbid it). To print `null`s nicely, I think you're supposed to use `%p` (another tip-of-the-cap to C's printf!).

I don't think that hacks in the hjson are the right approach though. I suggest fixing the code in the DV support macros to be compliant.